### PR TITLE
Android plugin V2 Migration

### PR DESCRIPTION
**Pull request by mitchross**
EDIT -@alr2413 is making good progress on a total rewrite of the plugin using federated plugins, the new approach to plugins in flutter. This is moving to draft for now. 

Flutter plugins require a v2 migration to avoid deprecation. https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration.

https://medium.com/flutter/modern-flutter-plugin-development-4c3ee015cf5a

### CODE REVIEW

**Comment by mitchross** [comment]
@solid-software This is ready to go, but I need a few people to test.

**Comment by pharshdev** [comment]
This is great, thanks. Can you take a look into [these issues](https://github.com/solid-software/flutter_vlc_player/pull/104#issuecomment-696665552)? I just tried the example code in this PR and sadly the issues still persist.

**Comment by mitchross** [comment]
@pharshdev @alr2413 Are looking into these... thanks!

File: android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayerPlugin.java [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
    */
   public static void registerWith(Registrar registrar) {
-    registrar.platformViewRegistry().registerViewFactory(
-            "flutter_video_plugin/getVideoView",
-            new FlutterVideoViewFactory(registrar.messenger(), registrar)
-    );
+    registrar
+            .platformViewRegistry()
+            .registerViewFactory(
+                    "plugins.flutter.io/webview",
```
Comment (alr2413) [inline android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayerPlugin.java:21]: this should be "flutter_video_plugin/getVideoView".

File: example/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -31,9 +31,8 @@
                  until Flutter renders its first frame. It can be removed if
                  there is no splash screen (such as the default splash screen
                  defined in @style/LaunchTheme). -->
-            <meta-data
-                android:name="io.flutter.app.android.SplashScreenUntilFirstFrame"
-                android:value="true" />
+            <meta-data android:name="flutterEmbedding" android:value="2"/>
```
Comment (alr2413) [inline example/android/app/src/main/AndroidManifest.xml:34]: I've read on flutter documentation this meta tag should be under application tag not activity.

**Review by alr2413 — COMMENTED** [review]
Please check my comments.

**Comment by alr2413** [comment]
During my review, I also found a problem that there is no "break" command after "MediaPlayer.Event.EndReached" case.
if you can please fix it.
thanks

**Comment by mitchross** [comment]
> "MediaPlayer.Event.EndReached

resolved, great catch.

File: android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayerPlugin.java (hunk 1 above)
Comment (vishal-akg) [reply android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayerPlugin.java]: I tested all the latest changes you commits you have made in the android files. 
release 4.0.1 is lags and loses state on orientation change or widget size changes.
comparing it with yours, your code make video screen black after orientation change.

Comment (alr2413) [reply android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayerPlugin.java]: That's because there is no runnable closure.

**Comment by mitchross** [comment]
@alr2413 is making good progress on a total rewrite of the plugin using federated plugins, the new approach to plugins in flutter. This is moving to draft for now.

**Pull request closed without merging**